### DISCUSSION TITLE
Get method with ? to receive parameter

### DISCUSSION THREAD

**Discussion by dileepgopal** [TypedProject/ts-express-decorators#1549] [Q&A] [answered]
Hello All,
Can anybody suggest a solution to receive query parameters with ? in the URL
My request URL is http://localhost:8080/method?q=paramvalue
If the controller method decorator is @Get("/method/:q") I am getting 404 error.

Your help would be much appreciated

Thanks,
Dileep G

**Comment by dileepgopal** [comment, marked as answer]
I found the solution as defined the controller get decorator as @Get("/method") and accessed the query param in the method as req.query.q
If any other better solution please suggest

**Reply by Romakita** [reply]
Hello @dileepgopal,
.... Have you read the documentation? this the base of the Controller feature.

```typescript
@Controller("/")
class MyCtrl {
  @Get('/path')
  get(@QueryParams('q') q: string) {
   console.log(q)
  }
}
```
http://localhost:8080/path?q=paramvalue

**Reply by dileepgopal** [reply]
Hello @Romakita 
Apologies for not showing patience to read the documentation.
Your code snippet worked for me

Thanks,
Dileep G

**Reply by Romakita** [reply]
;) no worries
